swarm/src/connection.rs

```diff
@@ -286,24 +301,35 @@ where
                 Poll::Ready(ConnectionHandlerEvent::ReportRemoteProtocols(
                     ProtocolSupport::Added(protocols),
                 )) => {
-                    if let Some(added) =
-                        ProtocolsChange::add(remote_supported_protocols, &protocols)
-                    {
-                        handler.on_connection_event(ConnectionEvent::RemoteProtocolsChange(added));
-                        remote_supported_protocols.extend(protocols);
+                    let added = protocols
+                        .into_iter()
+                        .filter(|i| remote_supported_protocols.insert(i.clone()))
```

Comment:
Why is the `.clone` needed here?

Reply:
since I need to insert the I and also send it to the connection handler